pkg/agent/multicast/mcast_discovery.go

```diff
@@ -242,8 +237,8 @@ func parseIGMPPacket(pkt protocol.Ethernet) (protocol.IGMPMessage, error) {
 	}
 }
 
-func newSnooper(ofClient openflow.Client, ifaceStore interfacestore.InterfaceStore, eventCh chan *mcastGroupEvent) *IGMPSnooper {
-	d := &IGMPSnooper{ofClient: ofClient, ifaceStore: ifaceStore, eventCh: eventCh}
+func newSnooper(ofClient openflow.Client, ifaceStore interfacestore.InterfaceStore, eventCh chan *mcastGroupEvent, queryInterval time.Duration) *IGMPSnooper {
+	d := &IGMPSnooper{ofClient: ofClient, ifaceStore: ifaceStore, eventCh: eventCh, queryInterval: queryInterval, mcastGroupTimeout: 3 * queryInterval}
```

Comment:
why it needs queryInterval and mcastGroupTimeout?

Reply:
`queryInterval` is need in generateIGMPQueryPacket, `mcastGroupTimeout` is removed since it is not needed here.